.github/workflows/rust.yml

```diff
@@ -7,12 +7,24 @@ jobs:
     name: Build and Test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: actions-rs/cargo@v1
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Build
+      uses: actions-rs/cargo@v1
       with:
         command: build
-        args: --manifest-path=calyx/Cargo.toml
-    - uses: actions-rs/cargo@v1
+    - name: Test
+      uses: actions-rs/cargo@v1
       with:
         command: test
-        args: --manifest-path=calyx/Cargo.toml
+  format:
```

Comment:
Is there a reason to separate this into a different job?

Reply:
I don't want to tie the testing job with formatting checks. Formatting is only needed at merge. Testing is useful at each commit.